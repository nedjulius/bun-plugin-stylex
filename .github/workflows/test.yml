name: 'Test'

on:
  push:
    branches:
      - 'main'
    pull_request:

jobs:
  main:
    runs-on: ubuntu-latest
    strategy:
      matrix:
        node-version: [20.x]
    steps:
      - uses: actions/checkout@v4
      - uses: oven-sh/setup-bun@v1
      - uses: actions/setup-node@v4
        with:
          node-version: '${{ matrix.node-version }}'
      - run: bun install
      - run: npm run test
