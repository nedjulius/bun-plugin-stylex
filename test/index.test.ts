import path from "node:path";
import { describe, it, expect } from "bun:test";
import createStylexPlugin from "../src";

async function build() {
  const [stylexPlugin, generateCSS] = createStylexPlugin();

  const output = await Bun.build({
    entrypoints: [path.resolve(__dirname, "fixtures/index.ts")],
    external: ["@stylexjs/stylex"],
    minify: false,
    plugins: [stylexPlugin],
  });
  const css = await generateCSS();

  return { output, css };
}

describe("bun-plugin-stylex", () => {
  it("base", async () => {
    const { output, css } = await build();

    expect(output).toMatchSnapshot();
    expect(css).toMatchSnapshot();
  });
});
