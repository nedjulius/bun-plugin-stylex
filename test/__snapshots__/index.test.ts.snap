// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`bun-plugin-stylex base 1`] = `
{
  "logs": [],
  "outputs": [
    BuildArtifact (entry-point) {
      path: "./index.js",
      loader: "ts",
      kind: "entry-point",
      hash: "7700c8a36a613cb7",
      Blob (124 bytes) {
        type: "text/javascript;charset=utf-8"
      },
      sourcemap: null
    },
  ],
  "success": true,
}
`;

exports[`bun-plugin-stylex base 2`] = `
"@keyframes xekv6nw-B{0%{opacity:0;}100%{opacity:1;}}
.x1ojj28o:not(#\#){animation:xekv6nw-B}
.x7z7khe:not(#\#){padding:10px}"
`;
